internal/settings/config_provider.go

```diff
@@ -28,27 +28,19 @@ import (
 	"sort"
 	"strconv"
 	"strings"
-	"sync"
 
 	"github.com/uber/prototool/internal/strs"
 	"go.uber.org/zap"
 	"gopkg.in/yaml.v2"
 )
 
 type configProvider struct {
```

Comment:
nit: Now that the cache is removed, it looks like we don't actually need this type. The `logger` is never used and all of the `configProvider` methods don't access any of the type's attributes. In other words, all of the methods could be package-level functions now. 

Removing this later might become more relevant depending on the direction we take https://github.com/uber/prototool/issues/93. If we only ever need one `prototool.yaml`, simple package-level functions might make more sense since we only need the single configuration struct.

Reply:
I still want to keep this abstraction for now, but we can revisit after #93.